src/zarr/config.py

```diff
@@ -2,16 +2,52 @@
 
 from typing import Any, Literal, cast
 
-from donfig import Config
+from donfig import Config as DConfig
+
+
+class BadConfigError(ValueError):
+    _msg = "bad Config: %r"
+
+
+class Config(DConfig):  # type: ignore[misc]
+    """Will collect configuration from config files and environment variables
+
+    Example environment variables:
+    Grabs environment variables of the form "DASK_FOO__BAR_BAZ=123" and
+    turns these into config variables of the form ``{"foo": {"bar-baz": 123}}``
+    It transforms the key and value in the following way:
+
+    -  Lower-cases the key text
+    -  Treats ``__`` (double-underscore) as nested access
+    -  Calls ``ast.literal_eval`` on the value
+
+    """
+
+    def reset(self) -> None:
+        self.clear()
+        self.refresh()
+
 
 config = Config(
-    "zarr",
+    "zarr_python",
     defaults=[
         {
             "array": {"order": "C"},
             "async": {"concurrency": None, "timeout": None},
-            "codec_pipeline": {"batch_size": 1},
             "json_indent": 2,
+            "codec_pipeline": {"name": "BatchedCodecPipeline", "batch_size": 1},
+            "codecs": {
+                "blosc": {"name": "BloscCodec"},
+                "gzip": {"name": "GzipCodec"},
+                "zstd": {"name": "ZstdCodec"},
+                "bytes": {"name": "BytesCodec"},
+                "endian": {"name": "BytesCodec"},  # compatibility with earlier versions of ZEP1
+                "crc32c": {"name": "Crc32cCodec"},
+                "sharding_indexed": {"name": "ShardingCodec"},
+                "transpose": {"name": "TransposeCodec"},
+            },
+            "buffer": {"name": "Buffer"},
+            "ndbuffer": {"name": "NDBuffer"},
```

Comment:
I'm a bit confused on how these entries interact with our entrypoints. Can we document this somewhere? Here or in registry.py would be good candidate locations.

Reply:
I've added some documentation: https://github.com/zarr-developers/zarr-python/pull/1982/commits/96271571c82cc87c09907318dbfc33215a282da0
Not sure about the level of detail, I'm open to feedback and discussion

Reply:
Entry points are great to register options. The configs here are meant to select of the options.